torch/_inductor/ir.py

```diff
@@ -1326,8 +1327,79 @@ def create_multilayer_existing_ranges(
         )
 
 
-def num_reduction_outputs(reduction_type):
-    return 3 if "welford" in reduction_type else 1
+class MultiOutputReduction(Reduction):
+    output_index: int
+
+    def __init__(
+        self,
+        device,
+        dst_dtype,
+        inner_fn,
+        ranges,
+        reduction_ranges,
+        reduction_type,
+        src_dtype,
+        reduction_hint,
+        output_index,
+    ):
+        super().__init__(
+            device,
+            dst_dtype,
+            inner_fn,
+            ranges,
+            reduction_ranges,
+            reduction_type,
+            src_dtype,
+            reduction_hint,
+        )
+        self.output_index = output_index
+
+    def store_reduction(self, output_name, indexer, vars, reduction_vars):
```

Comment:
This is just creating multiple identical reductions and indexing the output? Can we make welfordreduction also use this?